Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -259,7 +259,7 @@ bool AuxiliaryProcessProxy::sendMessage(UniqueRef<IPC::Encoder>&& encoder, Optio
 
     if (asyncReplyHandler && canSendMessage() && shouldStartProcessThrottlerActivity == ShouldStartProcessThrottlerActivity::Yes) {
         auto completionHandler = WTFMove(asyncReplyHandler->completionHandler);
-        asyncReplyHandler->completionHandler = [activity = throttler().quietBackgroundActivity(description(encoder->messageName())), completionHandler = WTFMove(completionHandler)](IPC::Decoder* decoder) mutable {
+        asyncReplyHandler->completionHandler = [activity = throttler().foregroundActivity(description(encoder->messageName())), completionHandler = WTFMove(completionHandler)](IPC::Decoder* decoder) mutable {
```

Comment:
2 issues here:
1. I don't think it is OK to take a foreground activity unconditionally. It should be on a case by case basis. If I send an async IPC to a background tab, e.g. to update some settings, there is no reason to give that background process *foreground* priority.
2. You want from a quiet activity to a non-quiet one, which means a lot of extra logging, which is bad for perf and power.

Reply:
That's a very good point, Chris! I will remove the changes related to taking foreground activities during message sending and page loading, and only address the process suspension issue on launch in this PR.